pkg/core/blockchain.go

```diff
@@ -2536,6 +2536,32 @@ func (bc *Blockchain) verifyAndPoolTx(t *transaction.Transaction, pool *mempool.
 	return nil
 }
 
+// CalculateAttributesFee returns network fee for all transaction attributes that should be
+// paid according to native Policy.
+func (bc *Blockchain) CalculateAttributesFee(tx *transaction.Transaction) int64 {
+	var (
+		feeCache map[transaction.AttrType]int64
```

Comment:
I'd not bother with caching here, leave it to the Policy.

Reply:
Policy will take read lock each time, and it adds some overhead. This `feeCache` is nil most of the time (transactions with attributes are quite rear). And in case of multiple attributes of the same time it will be a set of queries to RLock. Are you sure?